lib/jobs/upsert-node-job.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+module.exports = UpsertNodeAttributeFactory;
+
+var di = require('di');
+
+di.annotate(UpsertNodeAttributeFactory, new di.Provide('Job.Upsert.Node.Attribute'));
+di.annotate(UpsertNodeAttributeFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Services.Waterline',
+    'Logger'
+    )
+);
+
+function UpsertNodeAttributeFactory(BaseJob, util, waterline, Logger) {
+    var logger = Logger.initialize(UpsertNodeAttributeFactory);
+
+    function UpsertNodeAttribute(options, context, taskId) {
+        UpsertNodeAttribute.super_.call(this, logger, options, context, taskId);
+        this.nodeId = this.options.target;
```

Comment:
@zhangy42 I think @yyscamper's suggestion is good. This way it will be easier to add the task to existing workflows run against a specific node (and invoked via the `/api/2.0/nodes/<id>/workflows` api) without having to manually specify an ID in the workflow options. This is how we handle workflows that run against nodes in most other cases.